File: app/src/main/java/funnybrain/hsu/idv/taipeipark/restful/RestApiService.java
package funnybrain.hsu.idv.taipeipark.restful;

import funnybrain.hsu.idv.taipeipark.restful.model.WrappedData;
import retrofit2.http.GET;
import retrofit2.http.Query;
import rx.Observable;

public interface RestApiService {
    String API_ENDPOINT = "http://data.taipei/";

    @GET("opendata/datalist/apiAccess")
    Observable<WrappedData> getData(@Query("scope") String scope, @Query("rid") String rid);
}
// http://data.taipei/opendata/datalist/apiAccess?scope=resourceAquire&rid=bf073841-c734-49bf-a97f-3757a6013812

File: app/src/main/java/funnybrain/hsu/idv/taipeipark/restful/RestApiService.kt
package funnybrain.hsu.idv.taipeipark.restful

import funnybrain.hsu.idv.taipeipark.data.model.WrappedData
import retrofit2.http.GET
import retrofit2.http.Query
import rx.Observable

interface RestApiService {

    @GET("opendata/datalist/apiAccess")
    fun getData(@Query("scope") scope: String, @Query("rid") rid: String): Observable<WrappedData>

    companion object { // https://kotlinlang.org/docs/reference/object-declarations.html#companion-objects
        val API_ENDPOINT = "http://data.taipei/"
    }
}
// http://data.taipei/opendata/datalist/apiAccess?scope=resourceAquire&rid=bf073841-c734-49bf-a97f-3757a6013812